src/ui/Combobox/ComboboxValues.ts

```diff
@@ -0,0 +1,189 @@
+      $$(element).on('mouseleave', () => (this.mouseIsOverValue = false));
+      $$(element).on('click', this.onValueClick.bind(this));
+    });
+  }
+
+  private onValueClick(e: MouseEvent) {
+    if (!this.combobox.options.selectValueOnClick) {
+      return;
+    }
+
+    const target = <HTMLElement>e.target;
+    const targetElement = !$$(target).hasClass('coveo-combobox-value') ? $$(target).parent('coveo-combobox-value') : target;
+
+    if (!targetElement) {
+      return;
+    }
+
+    const value = find(this.values, ({ element }) => element.getAttribute('id') === targetElement.getAttribute('id'));
+    value && this.combobox.options.onSelectValue(value);
+    this.combobox.clearAll();
+  }
+
+  private updateAccessibilityAttributes() {
+    const activeDescendant = this.keyboardActiveValue ? this.keyboardActiveValue.element.getAttribute('id') : '';
+
+    this.combobox.updateAccessibilityAttributes({
+      activeDescendant,
+      expanded: this.hasValues()
+    });
+  }
+
+  public clearValues() {
+    this.mouseIsOverValue = false;
+    this.resetKeyboardActiveValue();
+    $$(this.element).empty();
+    $$(this.element).hide();
+    this.values = [];
+    this.updateAccessibilityAttributes();
+  }
+
+  private setKeyboardActiveValue(value: IComboboxValue) {
+    this.keyboardActiveValue = value;
+    this.activateFocusOnValue(this.keyboardActiveValue);
+  }
+
+  private resetKeyboardActiveValue() {
+    if (!this.keyboardActiveValue) {
+      return;
+    }
+
+    this.deactivateFocusOnValue(this.keyboardActiveValue);
+    this.keyboardActiveValue = null;
+  }
+
+  private activateFocusOnValue({ element }: IComboboxValue) {
+    $$(element).addClass('coveo-focused');
+    element.setAttribute('aria-selected', 'true');
+  }
+
+  private deactivateFocusOnValue({ element }: IComboboxValue) {
+    $$(element).removeClass('coveo-focused');
+    element.setAttribute('aria-selected', 'false');
+  }
+
+  public selectActiveValue() {
+    if (!this.keyboardActiveValue) {
+      return;
+    }
+
+    this.combobox.options.onSelectValue(this.keyboardActiveValue);
+    this.combobox.clearAll();
+  }
+
+  public moveActiveValueDown() {
+    if (!this.hasValues()) {
+      return;
+    }
+
+    const nextActiveValue = this.nextOrFirstValue;
+    this.resetKeyboardActiveValue();
+    this.setKeyboardActiveValue(nextActiveValue);
+    this.updateAccessibilityAttributes();
+  }
+
+  public moveActiveValueUp() {
+    if (!this.hasValues()) {
+      return;
+    }
+
+    const previousActiveValue = this.previousOrLastValue;
+    this.resetKeyboardActiveValue();
+    this.setKeyboardActiveValue(previousActiveValue);
+    this.updateAccessibilityAttributes();
+  }
+
+  private get nextOrFirstValue() {
+    if (!this.keyboardActiveValue) {
+      return this.values[0];
+    }
+
+    const nextValueIndex = this.values.indexOf(this.keyboardActiveValue) + 1;
+    return nextValueIndex < this.values.length ? this.values[nextValueIndex] : this.values[0];
```

Comment:
I think we could use the modulus operator here instead of an if
```
const nextValueIndex = (this.values.indexOf(this.keyboardActiveValue) + 1) %  this.values.length;
return this.values[nextValueIndex]
```

Reply:
🙌